addons/mail/static/src/models/audio.js

```diff
@@ -0,0 +1,117 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'Audio',
+    lifecycleHooks: {
+        _created() {
+            if (window.Audio === undefined) {
+                return;
+            }
+            this.update({ audio: new window.Audio() });
+        },
+    },
+    recordMethods: {
+        /**
+         * @param {Object} param0
+         * @param {boolean} [param0.loop] true if we want to make the audio loop, will only stop if stop() is called
+         * @param {float} [param0.volume] the volume percentage in decimal to play this sound.
+         *   If not provided, uses the default volume of this sound effect.
+         */
+        play({ loop = false, volume } = {}) {
+            if (this.messaging.isInQUnitTest) {
+                return;
+            }
+            if (!this.audio) {
+                return;
+            }
+            this.audio.pause();
+            this.audio.currentTime = 0;
+            this.audio.loop = loop;
+            this.audio.volume = volume !== undefined ? volume : this.defaultVolume;
+            Promise.resolve(this.audio.play()).catch(e => this._onAudioPlayError(e));
+        },
+        /**
+         * Resets the audio to the start of the track and pauses it.
+         */
+        stop() {
+            if (!this.audio) {
+                return;
+            }
+            this.audio.pause();
+            this.audio.currentTime = 0;
+        },
+        /**
+         * @private
+         * @param {DOMException} error
+         */
+        _onAudioPlayError(error) {
+            this.update({ audioPlayError: error });
+            // retry
```

Comment:
if there's a comment, at least explain why it is desirable to retry, and why only once (without catching error again)

Reply:
Also note that a 2nd attempt may fail again. `Audio.play()` needs at least one active user interaction on the page in order to work.

Reply:
I updated the comment to explain why retrying might be useful 👌

Reply:
the comment from alex is still important though, we might NOT want to fallback to mp3 if the error was just because the user didn't interact with the page yet

Reply:
My comment was more a FYI, but it's a good idea not to mistakenly fallback to mp3 when in this situation.
Let's see whether there's this info in the payload of `error` -- and if so -- don't fallback when this is due to lack of user interaction.

If there's none, I wouldn't advice doing something like intercepting all user interactions so that there's a trace in model that user make at least one user interaction.

Reply:
> the comment from alex is still important though, we might NOT want to fallback to mp3 if the error was just because the user didn't interact with the page yet

The fallback is only done if the error is NotSupportedError. AFAIK the error triggered when the user has not yet interacted with the page is NotAllowedError, so it shouldn't be a problem.